### PR TITLE
Add support for Profile editing

### PR DESCRIPTION
**Pull request by HugoMaster**

### CODE REVIEW

**Comment by nicolai86** [comment]
Oh, why a separate PR for this? can you unify #1 and #2 ?

**Comment by HugoMaster** [comment]
Ok, closed prev pr, and this one covers both.

File: android/src/main/java/rnxmpp/service/XmppServiceSmackImpl.java [hunk 1]
```diff
@@ -217,4 +221,28 @@ public void reconnectionFailed(Exception e) {
 
     }
 
+    @Override
+    public void editProfile(final ReadableMap params, String avatar){
```
Comment (nicolai86) [inline android/src/main/java/rnxmpp/service/XmppServiceSmackImpl.java]: why isn't avatar marked as `final` ?

File: android/src/main/java/rnxmpp/service/XmppService.java [hunk 2]
```diff
@@ -33,4 +34,7 @@
 
     @ReactMethod
     void sendStanza(String stanza);
+
+    @ReactMethod
+    void editProfile(final ReadableMap params, String avatar);
```
Comment (nicolai86) [inline android/src/main/java/rnxmpp/service/XmppService.java]: see below

File: android/src/main/java/rnxmpp/RNXMPPModule.java [hunk 3]
```diff
@@ -77,4 +78,10 @@ public void fetchRoster() {
     public void sendStanza(String stanza) {
         this.xmppService.sendStanza(stanza);
     }
+
+    @Override
+    @ReactMethod
+    public void editProfile(final ReadableMap options, String avatar) {
```
Comment (nicolai86) [inline android/src/main/java/rnxmpp/RNXMPPModule.java]: ❤️

File: RNXMPP/RNXMPPService.m [hunk 4]
```diff
@@ -356,7 +357,7 @@ - (void)xmppStream:(XMPPStream *)sender didReceiveTrust:(SecTrustRef)trust
     // The delegate method should likely have code similar to this,
     // but will presumably perform some extra security code stuff.
     // For example, allowing a specific self-signed certificate that is known to the app.
-
```
Comment (nicolai86) [inline RNXMPP/RNXMPPService.m]: can we get rid of the non-functional changes?

File: RNXMPP/RNXMPPService.m [hunk 5]
```diff
@@ -462,13 +463,13 @@ - (void)xmppRoster:(XMPPRosterMemoryStorage *)sender
 }
 
 - (void)xmppRoster:(XMPPRosterMemoryStorage *)sender
-    didRemoveResource:(XMPPResourceMemoryStorageObject *)resource withUser:(XMPPUserMemoryStorageObject *)user {
+ didRemoveResource:(XMPPResourceMemoryStorageObject *)resource withUser:(XMPPUserMemoryStorageObject *)user {
```
Comment (nicolai86) [inline RNXMPP/RNXMPPService.m]: same. can we only have changes related to functionality in the PR?

File: RNXMPP/RNXMPPService.m [hunk 6]
```diff
@@ -462,13 +463,13 @@ - (void)xmppRoster:(XMPPRosterMemoryStorage *)sender
...
 
     DDLogVerbose(@"%@: %@", THIS_FILE, THIS_METHOD);
 }
 
 - (void)xmppRoster:(XMPPRosterMemoryStorage *)sender
-    didUpdateResource:(XMPPResourceMemoryStorageObject *)resource withUser:(XMPPUserMemoryStorageObject *)user {
+ didUpdateResource:(XMPPResourceMemoryStorageObject *)resource withUser:(XMPPUserMemoryStorageObject *)user {
```
Comment (nicolai86) [inline RNXMPP/RNXMPPService.m]: bump

File: RNXMPP/RNXMPPService.m [hunk 7]
```diff
@@ -557,4 +558,24 @@ -(void)fetchRoster {
     [xmppRoster fetchRoster];
 }
 
+- (void)editProfile:(NSDictionary *)params avatar:(NSString *)avatar{
+
+    NSXMLElement *vCardXML = [NSXMLElement elementWithName:@"vCard" xmlns:@"vcard-temp"];
+    XMPPvCardTemp *newvCardTemp = [XMPPvCardTemp vCardTempFromElement:vCardXML];
+
+    //Set params
```
Comment (nicolai86) [inline RNXMPP/RNXMPPService.m]: please remove this comment

File: RNXMPP/RNXMPPService.m [hunk 8]
```diff
@@ -557,4 +558,24 @@ -(void)fetchRoster {
...
+    for(NSString *key in params.allKeys){
+        [newvCardTemp setValue:[params objectForKey:key] forKey:key];
+    }
+
+    //Set avatar image
```
Comment (nicolai86) [inline RNXMPP/RNXMPPService.m]: same

File: index.js [hunk 9]
```diff
@@ -101,6 +101,10 @@ class XMPP {
             React.NativeModules.RNXMPP.disconnect();
         }
     }
+
+    editProfile(params, avatar) {
+        RNXMPP.editProfile(params, avatar);
```
Comment (nicolai86) [inline index.js]: ❤️

File: RNXMPP/RNXMPPService.m (hunk 4 above)
Comment (HugoMaster) [inline RNXMPP/RNXMPPService.m]: Dont know why github see this, i dont have those changes while commiting... i will try to revert and maybe make clear commits.

**Comment by HugoMaster** [comment]
@nicolai86 how about now : )?

File: android/src/main/java/rnxmpp/service/XmppServiceSmackImpl.java [hunk 10]
```diff
@@ -217,4 +221,28 @@ public void reconnectionFailed(Exception e) {
 
     }
 
+    @Override
+    public void editProfile(ReadableMap params, String avatar){
```
Comment (nicolai86) [inline android/src/main/java/rnxmpp/service/XmppServiceSmackImpl.java]: I think both parameters should be final:

```public void editProfile(final ReadableMap params, final String avatar) { ```

File: android/src/main/java/rnxmpp/service/XmppService.java [hunk 11]
```diff
@@ -33,4 +34,7 @@
 
     @ReactMethod
     void sendStanza(String stanza);
+
+    @ReactMethod
+    void editProfile(ReadableMap params, String avatar);
```
Comment (nicolai86) [inline android/src/main/java/rnxmpp/service/XmppService.java]: see below

**Comment by nicolai86** [comment]
apart from the `final` LGTM. Before merging, can you create a new branch & make an upstream PR ? I think a new branch helps keeping our internal discussions separate from upstream discussions

**Pull request merged**